### ISSUE TITLE
Test (testcsvparser) fails on openSUSE Tumbleweed

### ISSUE THREAD

**Issue by plusky** [keepassxreboot/keepassxc#667]
Test fail when building keepassxc 2.2.0 on openSUSE Tumbleweed:
```
[   55s] 16/25 Test #19: testcsvparser ....................***Failed    0.11 sec
[   55s] ********* Start testing of TestCsvParser *********
[   55s] Config: Using QtTest library 5.9.0, Qt 5.9.0 (x86_64-little_endian-lp64 shared (dynamic) release build; by GCC 7.1.1 20170530 [gcc-7-branch revision 248621])
[   55s] PASS   : TestCsvParser::initTestCase()
[   55s] FAIL!  : TestCsvParser::testUnicode() 't.at(0).at(0) == "€1"' returned FALSE. ()
[   55s]    Loc: [/home/abuild/rpmbuild/BUILD/keepassxc-2.2.0/tests/TestCsvParser.cpp(333)]
[   55s] PASS   : TestCsvParser::testLF()
[   55s] PASS   : TestCsvParser::testEmptyReparsing()
[   55s] PASS   : TestCsvParser::testSimple()
[   55s] PASS   : TestCsvParser::testEmptyQuoted()
[   55s] PASS   : TestCsvParser::testEmptyNewline()
[   55s] PASS   : TestCsvParser::testSeparator()
[   55s] PASS   : TestCsvParser::testCR()
[   55s] PASS   : TestCsvParser::testCRLF()
[   55s] XFAIL  : TestCsvParser::testMalformed() Bad format
[   55s]    Loc: [/home/abuild/rpmbuild/BUILD/keepassxc-2.2.0/tests/TestCsvParser.cpp(67)]
[   55s] WARNING: TestCsvParser::testMalformed() malformed string: (row,col) 2,2
[   55s] 
[   55s]    Loc: [/home/abuild/rpmbuild/BUILD/keepassxc-2.2.0/tests/TestCsvParser.cpp(69)]
[   55s] PASS   : TestCsvParser::testMalformed()
[   55s] PASS   : TestCsvParser::testQualifier()
[   55s] PASS   : TestCsvParser::testNewline()
[   55s] PASS   : TestCsvParser::testEmptySimple()
[   55s] XFAIL  : TestCsvParser::testMissingQuote() Bad format
[   55s]    Loc: [/home/abuild/rpmbuild/BUILD/keepassxc-2.2.0/tests/TestCsvParser.cpp(57)]
[   55s] WARNING: TestCsvParser::testMissingQuote() missing closing quote: (row,col) 2,1
[   55s] 
[   55s]    Loc: [/home/abuild/rpmbuild/BUILD/keepassxc-2.2.0/tests/TestCsvParser.cpp(59)]
[   55s] PASS   : TestCsvParser::testMissingQuote()
[   55s] PASS   : TestCsvParser::testComments()
[   55s] PASS   : TestCsvParser::testBackslashSyntax()
[   55s] XFAIL  : TestCsvParser::testReparsing() Wrong qualifier
[   55s]    Loc: [/home/abuild/rpmbuild/BUILD/keepassxc-2.2.0/tests/TestCsvParser.cpp(291)]
[   55s] PASS   : TestCsvParser::testReparsing()
[   55s] PASS   : TestCsvParser::testEmptyFile()
[   55s] PASS   : TestCsvParser::testQuoted()
[   55s] PASS   : TestCsvParser::testMultiline()
[   55s] PASS   : TestCsvParser::testColumns()
[   55s] PASS   : TestCsvParser::cleanupTestCase()
[   55s] Totals: 22 passed, 1 failed, 0 skipped, 0 blacklisted, 5ms
[   55s] ********* Finished testing of TestCsvParser *********
[   55s] 
[   55s] 17/25 Test #18: testtotp .........................   Passed    0.11 sec
[   55s] 18/25 Test #21: testentrysearcher ................   Passed    0.10 sec
[   55s] 19/25 Test #20: testrandom .......................   Passed    0.10 sec
[   55s] 20/25 Test #17: testentry ........................   Passed    0.11 sec
[   55s] 21/25 Test #22: testexporter .....................   Passed    0.10 sec
[   55s] 22/25 Test #23: testcsvexporter ..................   Passed    0.10 sec
[   55s]       Start 24: testykchallengeresponsekey
[   55s]       Start 25: testdatabase
[   55s] 23/25 Test #25: testdatabase .....................   Passed    0.10 sec
[   55s] 24/25 Test #24: testykchallengeresponsekey .......   Passed    0.10 sec
[   97s] 25/25 Test #16: testautotype .....................   Passed   42.70 sec
[   97s] 
[   97s] 96% tests passed, 1 tests failed out of 25
```

**Comment by phoerious** [comment]
Are you building a 32 or 64 bit binary?

**Comment by plusky** [comment]
64 bit one

**Comment by phoerious** [comment]
What is your language setting?

**Comment by plusky** [comment]
In build environment defaults should be C, but enforcing en_US.UTF-8 seems to fix the problem

**Comment by phoerious** [comment]
That's what I suspected from the output. Will need to have a look at it later.

**Comment by magkopian** [comment]
@plusky I had the same issue while trying to build my package for the new release of KeePassXC. The issue is caused by some tests related to the CSV parser that require UTF-8 support in order to work properly.

Personally, I forced `LC_ALL` to `C.UTF-8` instead of `C` and it fixed the problem. I tried `en_US.UTF-8` at first, but I was getting errors from Perl. While this works, I feel that this is more of a workaround than a proper solution. I think the package should be able to be built, regardless of the locale settings.
